### PR TITLE
Created new UploadProgress controller

### PR DESCRIPTION
**Pull request by dmitry-zezik**

### CODE REVIEW

File: src/resources.qrc [hunk 1]
```diff
@@ -1,6 +1,5 @@
 <RCC>
     <qresource prefix="/">
-        <!-- Pages -->
```
Comment (fpohtmeh-github) [inline src/resources.qrc:3]: qt wizards/editors aren't comment-friendly

File: src/qml/components/UploadProgressBar.qml [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    property var model: models.fileCloudUploader
+    property int fileIndex: model.currentIndex
+    property int filesCount: model.fileNames.length
+    property real fileUploadValue: model.currentProcessedBytes * 100 / model.currentTotalBytes
+
+    property real uploadHeight: 60
+    property real uploadOpacity: 1
+
+    property real uploadFrom: 0
+    property real uploadTo: 100
```
Comment (fpohtmeh-github) [inline src/qml/components/UploadProgressBar.qml:20]: Use model.currentTotalBytes for **to** and model.currentProcessedBytes for **value**
0..100 range invokes extra conversion

File: src/qml/components/UploadProgressBar.qml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+
+import "../theme"
+
+
+Rectangle {
+    id: uploadProgressBar
+
+    property var model: models.fileCloudUploader
```
Comment (fpohtmeh-github) [inline src/qml/components/UploadProgressBar.qml:11]: readonly, readonly)

File: src/qml/components/UploadProgressBar.qml [hunk 4]
```diff
@@ -0,0 +1,115 @@
+import QtQuick 2.12
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+
+import "../theme"
+
```
Comment (fpohtmeh-github) [inline src/qml/components/UploadProgressBar.qml:6]: One empty row is enough

File: src/qml/components/UploadProgressBar.qml [hunk 5]
```diff
@@ -0,0 +1,115 @@
+import QtQuick 2.12
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
```
Comment (fpohtmeh-github) [inline src/qml/components/UploadProgressBar.qml:3]: Include if you use it only

File: src/qml/components/UploadProgressBar.qml [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+    Column {
+        anchors {
+            left: parent.left
+            right: parent.right
+            leftMargin: Theme.margin
+            rightMargin: Theme.margin
+            verticalCenter: parent.verticalCenter
+        }
+        height: this.contentHeight
+        spacing: 10
```
Comment (fpohtmeh-github) [inline src/qml/components/UploadProgressBar.qml:73]: Theme.qml ;)

File: src/qml/components/UploadProgressBar.qml [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+    Column {
+        anchors {
+            left: parent.left
+            right: parent.right
+            leftMargin: Theme.margin
+            rightMargin: Theme.margin
+            verticalCenter: parent.verticalCenter
+        }
+        height: this.contentHeight
```
Comment (fpohtmeh-github) [inline src/qml/components/UploadProgressBar.qml:72]: Good practice to use "this", if something doesn't work without "this". Just leave contentHeight.
Optional fix

File: include/models/FileCloudUploader.h [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+signals:
+    void currentIndexChanged(const int &index);
+    void currentProcessedBytesChanged(const DataSize &bytes);
+    void currentTotalBytesChanged(const DataSize &bytes);
+    void fileNamesChanged(const QStringList &fileNames);
+
+private:
+    QStringList m_fileNames;
+    int m_currentIndex;
```
Comment (fpohtmeh-github) [inline include/models/FileCloudUploader.h:73]: It's kind of serious issue if int/bool/double members aren't initialized in constructor (or right here in header). They will have unknown value

File: src/models/FileCloudUploader.cpp [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+//  HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+//  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+//  IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+//  POSSIBILITY OF SUCH DAMAGE.
+//
+//  Lead Maintainer: Virgil Security Inc. <support@virgilsecurity.com>
+
+#include "models/FileCloudUploader.h"
+
+#include <QObject>
```
Comment (fpohtmeh-github) [inline src/models/FileCloudUploader.cpp:37]: If you included something in the class header, it's visible in cpp (and everywhere where header is included). C++ magic)

File: src/resources.qrc (hunk 1 above)
Comment (dmitry-zezik) [reply src/resources.qrc]: Already done

File: src/qml/components/UploadProgressBar.qml (hunk 2 above)
Comment (dmitry-zezik) [reply src/qml/components/UploadProgressBar.qml:20]: Yes. That is reasonable

File: src/qml/components/UploadProgressBar.qml (hunk 3 above)
Comment (dmitry-zezik) [reply src/qml/components/UploadProgressBar.qml:11]: +

File: src/qml/components/UploadProgressBar.qml (hunk 4 above)
Comment (dmitry-zezik) [reply src/qml/components/UploadProgressBar.qml:6]: +

File: src/qml/components/UploadProgressBar.qml (hunk 5 above)
Comment (dmitry-zezik) [reply src/qml/components/UploadProgressBar.qml:3]: +

File: src/qml/components/UploadProgressBar.qml (hunk 6 above)
Comment (dmitry-zezik) [reply src/qml/components/UploadProgressBar.qml:73]: +

File: src/qml/components/UploadProgressBar.qml (hunk 7 above)
Comment (dmitry-zezik) [reply src/qml/components/UploadProgressBar.qml:72]: +

File: include/models/FileCloudUploader.h (hunk 8 above)
Comment (dmitry-zezik) [reply include/models/FileCloudUploader.h:73]: Initialized in constructor with (-1)

File: src/models/FileCloudUploader.cpp (hunk 9 above)
Comment (dmitry-zezik) [reply src/models/FileCloudUploader.cpp:37]: +

File: src/qml/components/UploadProgressBar.qml (hunk 7 above)
Comment (dmitry-zezik) [reply src/qml/components/UploadProgressBar.qml:72]: It doesn't work without this

**Review by fpohtmeh-github — APPROVED** [review]

**Review by fpohtmeh-github — APPROVED** [review]

**Pull request merged**